app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/action/SelectQualityAction.kt

```diff
@@ -0,0 +1,85 @@
+package org.jellyfin.androidtv.ui.playback.overlay.action
+
+import android.content.Context
+import android.view.Gravity
+import android.view.View
+import android.widget.PopupMenu
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.constant.QualityProfiles
+import org.jellyfin.androidtv.preference.UserPreferences
+import org.jellyfin.androidtv.preference.UserPreferences.Companion.maxBitrate
+import org.jellyfin.androidtv.ui.playback.PlaybackController
+import org.jellyfin.androidtv.ui.playback.VideoQualityController
+import org.jellyfin.androidtv.ui.playback.overlay.CustomPlaybackTransportControlGlue
+import org.jellyfin.androidtv.ui.playback.overlay.LeanbackOverlayFragment
+import org.koin.java.KoinJavaComponent
+import org.koin.java.KoinJavaComponent.get
+
+class SelectQualityAction (
+	context: Context,
+	customPlaybackTransportControlGlue: CustomPlaybackTransportControlGlue,
+	playbackController: PlaybackController
+) : CustomAction(context, customPlaybackTransportControlGlue) {
+
+	private var previousQualitySelection = QualityProfiles.fromPreference(
+		get<UserPreferences>(UserPreferences::class.java)[maxBitrate]
```

Comment:
inject user preferences via constructor to avoid KoinKavaComponent

Reply:
Done.